sources/packages/api/src/services/application/application.service.ts

```diff
@@ -204,4 +210,23 @@ export class ApplicationService extends RecordDataModelService<Application> {
 
     return application?.offering;
   }
+  /**
+   * get applications of a institution location
+   * with PIR status required and completed.
+   * @param locationId location id .
+   * @returns student Application list.
+   */
+  async getAllLocationApplications(
+    locationId: number,
+  ): Promise<LocationApplications[]> {
```

Comment:
Please change the query to use Typeform.
If there is a particular reason to not use the ORM we can discuss it, otherwise, I believe that we should rely on ORM as a first choice to execute all queue DB queries.